Sources/TuistGenerator/Generator/BuildPhaseGenerator.swift

```diff
@@ -715,6 +724,42 @@ final class BuildPhaseGenerator: BuildPhaseGenerating {
         embedXPCServicesBuildPhase.files = pbxBuildFiles
     }
 
+    func generateEmbedPlugInsBuildPhase(
+        path: AbsolutePath,
+        target: Target,
+        graphTraverser: GraphTraversing,
+        pbxTarget: PBXTarget,
+        fileElements: ProjectFileElements,
+        pbxproj: PBXProj
+    ) throws {
+        let targetDependencies = graphTraverser.directLocalTargetDependencies(path: path, name: target.name).sorted()
+        let plugIns = targetDependencies.filter { $0.target.isEmbeddablePlugin() }
+        guard !plugIns.isEmpty else { return }
+
+        let embedPlugInsBuildPhase = PBXCopyFilesBuildPhase(
+            dstSubfolderSpec: .plugins,
+            name: "Embed PlugIns"
+        )
+        pbxproj.add(object: embedPlugInsBuildPhase)
+        pbxTarget.buildPhases.append(embedPlugInsBuildPhase)
+
+        let pbxBuildFiles: [PBXBuildFile] = plugIns.compactMap { graphTarget in
+            guard let fileReference = fileElements.product(target: graphTarget.target.name) else { return nil }
+
+            let pbxBuildFile = PBXBuildFile(
+                file: fileReference,
+                settings: ["ATTRIBUTES": ["CodeSignOnCopy", "RemoveHeadersOnCopy"]]
+            )
+            if target.supportsCatalyst {
+                pbxBuildFile.applyPlatformFilters([.catalyst])
+            }
```

Comment:
I’m not sure where it would be most appropriate to add the same logic for applying filters to "Target Dependencies", as changes to these two sections in Xcode are always synchronized. 

However, even without handling this, it won’t affect the product archived by Xcode.

Reply:
Alternatively, we could perform the validation during the linting phase.